scripts/create-kubestellar-demo-env.sh

```diff
@@ -84,35 +117,64 @@ checking_cluster() {
 
     done
 }
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
 ##########################################
 
 echo -e "\nStarting environment clean up..."
 echo -e "Starting cluster clean up..."
 
-cluster_clean_up "kind delete cluster --name kubeflex" &
-cluster_clean_up "kind delete cluster --name cluster1" &
-cluster_clean_up "kind delete cluster --name cluster2" &
+if [ "$k8s_platform" == "kind" ]; then
+    if command_exists "k3d"; then
+        cluster_clean_up "k3d cluster delete kubeflex" &
+        cluster_clean_up "k3d cluster delete cluster1" &
+        cluster_clean_up "k3d cluster delete cluster2" &
+        wait
+    fi
+    if command_exists "kind"; then
+        cluster_clean_up "kind delete cluster --name kubeflex" &
+        cluster_clean_up "kind delete cluster --name cluster1" &
+        cluster_clean_up "kind delete cluster --name cluster2" &
+        wait
+    fi
+fi
```

Comment:
Maybe the thrust of my earlier comments was not clear. I do not debate the need for cleanup of from both platforms. I was asking whether/why the _order of_ the cleanup matters.

Without the cleanup of `kind` clusters when creating `k3d` clusters, I get a failure due to port 9443 already being taken. Following is an excerpt from a trial.

```
Creating two k3d clusters to serve as example WECs
Creating KubeFlex cluster with SSL Passthrough
INFO[0000] portmapping '9443:443' targets the loadbalancer: defaulting to [servers:*:proxy agents:*:proxy] 
INFO[0000] Prep: Network                                
INFO[0000] Re-using existing network 'k3d-kubeflex' (2bce747eb25b10de393f0ce44d360ee9bf90b076d5d22de7166211e66f563170) 
INFO[0000] Created image volume k3d-kubeflex-images     
INFO[0000] Starting new tools node...                   
INFO[0000] Starting node 'k3d-kubeflex-tools'           
INFO[0001] Creating node 'k3d-kubeflex-server-0'        
INFO[0001] Creating LoadBalancer 'k3d-kubeflex-serverlb' 
INFO[0001] Using the k3d-tools node to gather environment information 
INFO[0001] HostIP: using network gateway 172.22.0.1 address 
INFO[0001] Starting cluster 'kubeflex'                  
INFO[0001] Starting servers...                          
INFO[0001] Starting node 'k3d-kubeflex-server-0'        
INFO[0004] All agents already running.                  
INFO[0004] Starting helpers...                          
INFO[0004] Starting node 'k3d-kubeflex-serverlb'        
ERRO[0004] Failed Cluster Start: Failed to add one or more helper nodes: runtime failed to start node 'k3d-kubeflex-serverlb': docker failed to start container for node 'k3d-kubeflex-serverlb': Error response from daemon: driver failed programming external connectivity on endpoint k3d-kubeflex-serverlb (dc3dd5564bd8cbae1d7c1b7193c1f860012e71e5f81501267af15467870e4333): Bind for 0.0.0.0:9443 failed: port is already allocated 
ERRO[0004] Failed to create cluster >>> Rolling Back    
INFO[0004] Deleting cluster 'kubeflex'                  
INFO[0004] Deleting 1 attached volumes...               
FATA[0004] Cluster creation FAILED, all changes have been rolled back! 
```

Reply:
thats correct, that is why both cleanups are handled here in this control block. Cleanup is done here to remove debris before trying to create new clusters of any sort. So there is no code change you are proposing, your just asking for info?

Reply:
The latest revision addresses my concern.